core/src/main/java/org/wso2/carbon/kernel/utils/Tokenizer.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.kernel.utils;
+
+/**
+ * Simple tokenizer class. Used to parse data.
+ * This class is taken from org.eclipse.osgi.framework.internal.core
+ *
+ * @since 5.0.0
+ */
+public class Tokenizer {
+    private char value[];
+    private int max;
+    private int cursor;
+    /**
+     * Construct the tokenizer from the given string
+     *
+     * @param value String
+     */
+    public Tokenizer(String value) {
+        this.value = value.toCharArray();
+        max = this.value.length;
+        cursor = 0;
+    }
+
+    private void skipWhiteSpace() {
+        char[] val = value;
+        int cur = cursor;
+
+        for (; cur < max; cur++) {
```

Comment:
Here we have taken org.eclipse.osgi.framework.internal.core.tokenizer class into our code base to get rid of unwanted dependencies. Thus the logic is intact from it's original version to make sure the code does not break.